UM/Operations/Operation.py

```diff
@@ -57,3 +57,11 @@ def push(self) -> None:
         # Because of circular dependency
         from UM.Application import Application
         Application.getInstance().getOperationStack().push(self)
+
+    def clear(self) -> None:
+        # Because of circular dependency
+        from UM.Application import Application
+        Application.getInstance().getOperationStack().clearStack()
```

Comment:
This is a bit weird that the operation object clears the stack, does that mean it will be done multiple times ? Wouldn't it be the job of the caller then ? Or maybe the operation should only remove itself instead.

Reply:
I don't think it'll be cleared multiple times, its only when an object of operation asks to clear the stack. I took the example from the function push just below this function. The push in operation stack also happens from one of the operation.